src/lib/structures/ModerationManager.ts

```diff
@@ -69,51 +70,54 @@ export class ModerationManager extends Collection<number, ModerationManagerEntry
 		if (this.size === 0) return null;
 
 		const minimumTime = Date.now() - (15 * Time.Second);
-		return this.reduce<ModerationManagerEntry>((prev, curr) => curr.flattenedUser === userID
+		return this.reduce<ModerationEntity | null>((prev, curr) => curr.userID === userID
 			? prev === null
 				? curr.createdTimestamp >= minimumTime ? curr : prev
 				: curr.createdTimestamp > prev.createdTimestamp ? curr : prev
-			: prev, undefined);
+			: prev, null);
 	}
 
 	public create(data: ModerationManagerCreateData) {
-		return new ModerationManagerEntry(this, { guild_id: this.guild.id, ...data });
+		return new ModerationEntity(data).setup(this);
 	}
 
-	public async fetch(id: number): Promise<ModerationManagerEntry | null>;
-	public async fetch(id: string | number[]): Promise<Collection<number, ModerationManagerEntry>>;
+	public async fetch(id: number): Promise<ModerationEntity | null>;
+	public async fetch(id: string | number[]): Promise<Cache<number, ModerationEntity>>;
 	public async fetch(id?: null): Promise<this>;
-	public async fetch(id?: string | number | number[] | null): Promise<ModerationManagerEntry | Collection<number, ModerationManagerEntry> | this | null> {
+	public async fetch(id?: string | number | number[] | null): Promise<ModerationEntity | Cache<number, ModerationEntity> | this | null> {
 		// Case number
 		if (typeof id === 'number') {
-			return super.get(id) || this._cache(await this.guild.client.queries.fetchModerationLogByCase(this.guild.id, id), CacheActions.None);
+			return super.get(id) || this._cache(await (await DbSet.connect()).moderations.find({ where: { guildID: this.guild.id, caseID: id } }), CacheActions.None);
 		}
 
 		// User id
 		if (typeof id === 'string') {
 			return this._count === super.size
-				? super.filter(entry => entry.user === id)
-				: this._cache(await this.guild.client.queries.fetchModerationLogsByUser(this.guild.id, id), CacheActions.None);
+				? super.filter(entry => entry.userID === id)
+				: this._cache(await (await DbSet.connect()).moderations.find({ where: { guildID: this.guild.id, userID: id } }), CacheActions.None);
 		}
 
 		if (Array.isArray(id) && id.length) {
-			return this._cache(await this.guild.client.queries.fetchModerationLogsByCases(this.guild.id, id), CacheActions.None);
+			return this._cache(await (await DbSet.connect()).moderations.find({ where: { guildID: this.guild.id, caseID: id } }), CacheActions.None);
 		}
 
 		if (super.size !== this._count) {
-			this._cache(await this.guild.client.queries.fetchModerationLogsByGuild(this.guild.id), CacheActions.Fetch);
+			this._cache(await (await DbSet.connect()).moderations.find({ where: { guildID: this.guild.id } }), CacheActions.Fetch);
 		}
 		return this;
 	}
 
 	public async count() {
 		if (this._count === null) {
-			this._count = await this.client.queries.fetchModerationCurrentCaseID(this.guild.id);
+			this._count = await (await DbSet.connect()).moderations.count({ where: { guildID: this.guild.id } });
```

Comment:
It pains me to see this... Is there really no better way to avoid `await (await)`?